src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs

```diff
@@ -52,46 +52,43 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(CompletionParams request
 
     public async Task<VSInternalCompletionList?> HandleRequestAsync(CompletionParams request, RazorRequestContext requestContext, CancellationToken cancellationToken)
     {
-        var documentContext = requestContext.DocumentContext;
-
-        if (request.Context is null || documentContext is null)
-        {
-            return null;
-        }
-
-        var sourceText = await documentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
-        if (!sourceText.TryGetAbsoluteIndex(request.Position, out var hostDocumentIndex))
+        if (request.Context is not VSInternalCompletionContext completionContext ||
+            requestContext.DocumentContext is not { } documentContext)
         {
             return null;
         }
 
-        if (request.Context is not VSInternalCompletionContext completionContext)
+        var autoShownCompletion = completionContext.InvokeKind != VSInternalCompletionInvokeKind.Explicit;
+        var options = _optionsMonitor.CurrentValue;
+        if (autoShownCompletion && !options.AutoShowCompletion)
         {
-            Debug.Fail("Completion context should never be null in practice");
             return null;
         }
 
-        var autoShownCompletion = completionContext.InvokeKind != VSInternalCompletionInvokeKind.Explicit;
-        if (autoShownCompletion && !_optionsMonitor.CurrentValue.AutoShowCompletion)
+        var sourceText = await documentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
ℹ️ I've reordered some of the checks to push fast checks ahead of checks that require async work.

Reply:
Do any of these changes apply to the cohost completion endpoint? Though even if they do, I'm fine if you want to leave that for me, I suspect I need to audit that, and compare to this endpoint anyway

Reply:
I didn't look actually look at the co-hosting endpoint. I was primarily looking at callers of `CompletionListProvider.GetCompletionListAsync(...)`. For co-hosting, that's `RemoteCompletionService`.

Reply:
The options checking would be in the endpoint, which is why I thought of it, but I just looked and the ordering of the checks there is fine.